include/knowhere/config.h

```diff
@@ -402,6 +406,10 @@ class BaseConfig : public Config {
             .description("search for top k similar vector.")
             .set_range(1, std::numeric_limits<CFG_INT>::max())
             .for_search();
+        KNOWHERE_CONFIG_DECLARE_FIELD(num_build_thread)
+            .set_default(-1)
```

Comment:
-1 is a valid value ?

Reply:
I need a default value to indicate user has not set this value, and '-1' gives this semantic.